docs/source/changelog.md

```diff
@@ -12,6 +12,326 @@ and as we merge [breaking changes in pull
+
+  TODO
+
+#### Security: breaking change to `*.networkPolicy.egress`
+
+If you have configured any of:
+
+- `hub.networkPolicy.egress`
+- `proxy.chp.networkPolicy.egress`
+- `proxy.traefik.networkPolicy.egress`
+- `singleuser.networkPolicy.egress`
+
+you must review your configuration as additional default egress routes have been added.
+Previously `...networkPolicy.egress` controlled all egress but a new properties `...networkPolicy.egressAllowRules` add additional egress rules by default.
+
+If you have configured `...networkPolicy.egress` for `hub`, `proxy.chp`,
+`proxy.traefik` or `singleuser` to restrict the permissions to establish
+outbound network connections, then this upgrade is likely to _escalate those
+permissions unless you revise your configuration_. The new configuration
+`...networkPolicy.egressAllowRules` are by default granting most of the egress
+permissions previously granted by default via the `...networkPolicy.egress`
+configuration, and `...networkPolicy.egress` are now by default not providing
+any permissions.
+
+If you for example had overridden the previously very permissive default value
+of `singleuser.networkPolicy.egress` to be less permissive, you should consider
+disabling all `singleuser.networkPolicy.egressAllowRules` like this
+to not risk escalating the permissions.
+
+```yaml
+singleuser:
+  networkPolicy:
+    egressAllowRules:
+      cloudMetadataServer: false
+      dnsPortsPrivateIPs: false
+      nonPrivateIPs: false
+      privateIPs: false
+```
+
+For more details, see the documentation on [Kubernetes Network Policies](netpol)
+and the configuration reference entries under
+[`...networkPolicy.egress`](schema_hub.networkPolicy.egress) and
+[`...networkPolicy.egressAllowRules`](schema_hub.networkPolicy.egressAllowRules).
+
+#### Upgrade instructions
+
+1. Adjust your configuration based on the security notice above if it applies to
+   your deployment.
+2. Prepare a user image with a modern version of JupyterHub installed. Install
+   `jupyterhub>=2.3.1` via pip or `jupyterhub-base>=2.3.1` via conda/mamba in
+   the image.
+   Update the configuration to use this new image, which is typically done via
+   `singleuser.image` or as part of `singleuser.profileList`.
+3. Read and adjust your configuration to the breaking changes below.
+
+#### Breaking changes
+
+These breaking changes have been made relative to the 1.\* series of Z2JH releases.
+
+##### JupyterHub 2 and related hub components
+
+Z2JH 2.0.0 upgrades JupyterHub to the 2.\* series, and also upgrades all hub components.
+If you are using any custom JupyterHub services, addons, API integrations, or extra configuration, you should review the breaking changes in the
+[JupyterHub 2.0.0 changelog](https://github.com/jupyterhub/jupyterhub/blob/2.3.1/docs/source/changelog.md#200).
+
+JupyterHub 2 uses an updated database schema.
+Z2JH 2.0.0 automatically handles the upgrade, but it will not be possible to downgrade to older releases after this.
```

Comment:
```suggestion
JupyterHub 2 updates the database schema, which means a migration takes place when you upgrade JupyterHub.
Z2JH automatically handles the upgrade if you are using sqlite (`hub.db.type = 'sqlite-pvc', the default), but it may not be possible to downgrade to older releases after this.
When using sqlite, JupyterHub automatically creates a backup in the `hub-db` volume,
which can be restored manually if you need to downgrade.
```

Reply:
@minrk I've copied your suggestions to the new upgade guide: https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2803